ini.c

```diff
@@ -46,22 +46,34 @@ static int ini_property(struct ini *ini, char *key, char *value)
 	if (!ini->prop_cb)
 		return 0;
 
-	// If the value begins with 'xresource:` then query Xresources for the actual value.
-    if (strncmp(value, "xresource:", strlen("xresource:")) == 0) {
-        if (!ini->database) {
-            fatal("Unable to access Xresources.");
-            return -EINVAL;
-        }
-
-        char *resource;
-        if (xcb_xrm_resource_get_string(ini->database, value + strlen("xresource:"), NULL, &resource) == 0) {
-            value = strdup(resource);
```

Comment:
value is a pointer into a char array allocated on the stack, so the memory allocated by `strdup(resource)` is never free'd.

Reply:
For reference, the buffer is in `line.c:line_parse`